### DISCUSSION TITLE
How will Nuclei evolve in the future ? (ex : integration of Windows protocols)

### DISCUSSION THREAD

**Discussion by mxrch** [projectdiscovery/nuclei#1201] [Q&A] [answered]
Hello, thanks for this amazing tool, I can affirm that it gives 0 false positives.
I want to use Nuclei for a professionnal use, currently we use Nessus but I don't like it so much.
So I'm making a comparison list between Nuclei and Nessus, but here is the problem ;
I don't know where you want Nuclei to go, it looks primarely web-focused, but is has SSH templates, so :

1) Will it support more protocols in the future ? Like SMB, WinRM, RDP,...
2) Can we already create templates to test these protocols ? Not just looking at the banner but for example what if I want to make an Eternal Blue template, or something more sophisticated ? Isn't a need to create a scripting language like Nmap and Nessus does, or integrate Golang scripts ?

**Comment by ehsandeep** [comment, marked as answer]
Appreciate @mxrch for initiating this conversation and providing context; yes, we are working to support other protocols; currently, a couple are in development; additional details are available in this discussion - #1119.

**Reply by mxrch** [reply]
Thank you for the details about the protocols ! But I don't see an answer about complexe checks like for Eternal Blue, is there already a way to make template for them ? And what about a scripting language / Golang scripts ?

**Reply by ehsandeep** [reply]
@mxrch in the current form and capability, it can not be done, we have a PR (WIP) to support python like scripting - https://github.com/projectdiscovery/nuclei/pull/914 that will supports writing more complex checks.
